packages/components/src/components/domainproperties/dataset/DatasetPropertiesPanelFormElements.tsx

```diff
@@ -0,0 +1,280 @@
+                    onChange={onRadioChange}
+                >
+                    Participant ID, timepoint, and additional key field
+                </Radio>
+            </div>
+        );
+    }
+}
+
+interface DataRowUniquenessContainerProps {
+    model: DatasetModel;
+    onRadioChange: (e: any) => any;
+    onCheckBoxChange: (any) => void;
+    onSelectChange: (name, formValue, selected) => void;
+}
+
+export class DataRowUniquenessContainer extends React.PureComponent<DataRowUniquenessContainerProps> {
+
+    getDataRowSetting(model: DatasetModel) : number {
+        let dataRowSetting;
+
+        // participant id
+        if ((model.keyPropertyId === undefined || model.keyPropertyId === null) && model.isDemographicData) {
+            dataRowSetting = 0;
+        }
+        // participant id and timepoint
+        else if (model.keyPropertyId === undefined || model.keyPropertyId === null) {
+            dataRowSetting = 1;
+        }
+        // participant id, timepoint and additional key field
+        else {
+            dataRowSetting = 2;
+        }
+
+        return dataRowSetting;
+    }
+
+    getHelpTipElement(field: string) : JSX.Element {
+        return <> {getHelpTip(field)} </> as JSX.Element;
+    }
+
+    render() {
+        const { model, onRadioChange, onCheckBoxChange, onSelectChange } = this.props;
+        const domain = model.domain;
+
+        const dataRowSetting = this.getDataRowSetting(model);
+        const showAdditionalKeyField = dataRowSetting === 2;
+
+        const showAdditionalKeyFieldCls = showAdditionalKeyField ? "dataset_data_row_uniqueness_keyField_show margin-top" : "dataset_data_row_uniqueness_keyField_hide";
+
+        return (
+            <>
+                <SectionHeading
+                    title="Data Row Uniqueness"
+                    helpTipBody={() => this.getHelpTipElement("dataRowUniqueness")}
+                />
+
+                <DataRowUniquenessElements
+                    onRadioChange={onRadioChange}
+                    dataRowSetting={dataRowSetting}
+                />
+
+                <div className={showAdditionalKeyFieldCls}>
+                    <DatasetSettingsSelect
+                        name={"keyPropertyId"}
+                        label={"Additional Key Field"}
+                        selectOptions={domain.fields.toArray()}
+                        onSelectChange={onSelectChange}
+                        labelKey={"label"}
+                        valueKey={"propertyId"}
+                        selectedValue={model.keyPropertyId}
+                        disabled={!showAdditionalKeyField}
+                    />
+
+                    <div className='margin-top'>
+
+                        <Checkbox
+                            checked={model.keyPropertyManaged}
+                            onChange={onCheckBoxChange}
+                            id={"keyPropertyManaged"}
+                            disabled={!showAdditionalKeyField}
```

Comment:
See comment above, to match the functionality from the GWT designer, we'll also need to disable this checkbox if the selected additional key field is not a numeric or string type.